libcontainer/container_linux.go

```diff
@@ -779,3 +780,28 @@ func (c *linuxContainer) currentState() (*State, error) {
 	}
 	return state, nil
 }
+
+func (c *linuxContainer) Kill(sigStr string) error {
+	var sig uint64
+	sigN, err := strconv.ParseUint(sigStr, 10, 5)
+	if err != nil {
+		//The signal is not a number, treat it as a string (either like
+		//KILL" or like "SIGKILL")
+		syscallSig, ok := SignalMap[strings.TrimPrefix(sigStr, "SIG")]
+		if !ok {
+			return fmt.Errorf("Invalid signal: %s", sigStr)
+		}
+		sig = uint64(syscallSig)
+	} else {
+		sig = sigN
+	}
+
+	if sig == 0 {
+		return fmt.Errorf("Invalid signal: %s", sigStr)
+	}
+	state, err := c.State()
+	if err != nil {
+		return err
+	}
+	return syscall.Kill(state.InitProcessPid, syscall.Signal(sig))
```

Comment:
@marcosnils 
Though I've implemented the change for unit test under integration, I would like to mention one point. When we do make localtest, as the containers are getting started, once the parent process died out , automatically the corresponding name space process also dies. Even if I make the container to be killed, it will be killed automatically when the parent dies. I was just thinking will it be useful in unit testing ?
 @avagin 
My first PR request commits I've made similar to what you suggested. Later I moved to libcontainer.
I'm moving back the parsing logic to runc now and submit the commits